backend/Server/Server.java

```diff
@@ -0,0 +1,279 @@
+    public static String getServerContext() {
+        return serverContext;
+    }
+    
+    public static int getPort() {
+        return port;
+    }
+    
+    public static String getHostname() {
+        return hostname;
+    }
+    
+    public static void stop() {
+        try {
+            server.stop(serverCloseDelay);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+    
+    @Override
+    public void run() {
+        try {
+            this.server.createContext(this.serverContext, new ServerConnectionHandler());
+            this.server.start();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+    
+    public static void main(final String... args) throws IOException {
+        server = HttpServer.create(new InetSocketAddress(hostname, port), backlog);
+        server.createContext(serverContext, new ServerConnectionHandler());
+        server.setExecutor(null); // creates a default executor
+        server.start();
+        System.out.println(String.format("Server open on port %s", port));
+        System.out.println(String.format("Connect via: http://localhost:%s", port));
+        // testing
+        System.out.println(" ".repeat(5) + "Press enter to quit");
+        System.in.read();
+        System.out.println(String.format("Stopping server, waiting up to %s seconds...", serverCloseDelay));
+        // -----
+        server.stop(serverCloseDelay);
+        if (isRunning()) {
+            System.out.println(String.format("Server was not properly close on hostname %s, port %s", hostname, port));
+        } else {
+            System.out.println(String.format("Server close on port %s", port));
+        }
+    }
+    
+    private static String decodeUrlComponent(final String urlComponent) {
+        try {
+            return URLDecoder.decode(urlComponent, CHARSET.name());
+        } catch (final UnsupportedEncodingException e) {
+            return null;
+        }
+    }
+    
+    private static String getRequestFragment(final String rawRequestFragment) {
+        if (rawRequestFragment != null) {
+            return decodeUrlComponent(rawRequestFragment);
+        }
+        return null;
+    }
+    
+    private static Map<String, String> getRequestQueryParameters(final String rawRequestQuery) {
+        final Map<String, String> queryParameters = new LinkedHashMap<>();
+        if (rawRequestQuery != null) {
+            final String[] keyValues = rawRequestQuery.split("[&;]", -1);
+            for (final String keyValue : keyValues) {
+                final String[] parameter = keyValue.split("=", 2);
+                final String key = decodeUrlComponent(parameter[0]);
+                final String value = (parameter.length > 1) ? decodeUrlComponent(parameter[1]) : null;
+                queryParameters.putIfAbsent(key, value);
+            }
+        }
+        return queryParameters;
+    }
+    
+    // Helper function, remove before deployment

```

Comment:
Project deploy or PR?
If project deploy, make sure we have some reminder/note that **will be followed**.

Reply:
Project deploy, not PR. I try to keep these comments as consistently worded as possible so its easy to find all of these notes with a find/replace search.